test/src/main/java/io/strimzi/test/BaseITST.java

```diff
@@ -0,0 +1,140 @@
+    }
+
+    private void deleteClusterOperatorInstallFiles() {
+        TimeMeasuringSystem.setTestName(testClass, testClass);
+        TimeMeasuringSystem.startOperation(Operation.CO_DELETION);
+        for (Map.Entry<File, String> entry : clusterOperatorMap.entrySet()) {
+            LOGGER.info("Removing configuration file: {}", entry.getKey());
+            KUBE_CLIENT.clientWithAdmin().deleteContent(entry.getValue());
+        }
+        TimeMeasuringSystem.stopOperation(Operation.CO_DELETION);
+    }
+
+    private void createNamespaces(String useNamespace, List<String> namespaces) {
+        deploymentNamespaces = namespaces;
+        for (String namespace: namespaces) {
+            LOGGER.info("Creating namespace: {}", namespace);
+            KUBE_CLIENT.createNamespace(namespace);
+            KUBE_CLIENT.waitForResourceCreation("Namespace", namespace);
+            LOGGER.info("Namespace {} created", namespace);
+        }
+        LOGGER.info("Using namespace {}", useNamespace);
+        KUBE_CLIENT.namespace(useNamespace);
+    }
+
+    protected void createNamespaces(String useNamespace) {
+        createNamespaces(useNamespace, Collections.singletonList(useNamespace));
+    }
+
+    protected void deleteNamespaces() {
+        for (String namespace: deploymentNamespaces) {
+            LOGGER.info("Deleting namespace: {}", namespace);
+            KUBE_CLIENT.deleteNamespace(namespace);
+            KUBE_CLIENT.waitForResourceDeletion("Namespace", namespace);
+            LOGGER.info("Namespace {} deleted", namespace);
+        }
+        LOGGER.info("Using namespace {}", CLUSTER.defaultNamespace());
+        KUBE_CLIENT.namespace(CLUSTER.defaultNamespace());
+    }
+
+    protected void createCustomResources(List<String> resources) {
+        deploymentResources = resources;
+        for (String resource : resources) {
+            LOGGER.info("Creating resources {}", resource);
+            KUBE_CLIENT.clientWithAdmin().create(resource);
+        }
+    }
+
+    protected void deleteCustomResources() {
+        for (String resource : deploymentResources) {
+            LOGGER.info("Deleting resources {}", resource);
+            KUBE_CLIENT.delete(resource);
+        }
+    }
+
+    protected void prepareEnvForOperator(String clientNamespace, List<String> namespaces, List<String> resources) {
+        createNamespaces(clientNamespace, namespaces);
+        createCustomResources(resources);
+        applyClusterOperatorInstallFiles();
+    }
+
+    protected void prepareEnvForOperator(String clientNamespace, List<String> resources) {
+        prepareEnvForOperator(clientNamespace, Collections.singletonList(clientNamespace), resources);
+    }
+
+    protected void prepareEnvForOperator(String clientNamespace) {
+        prepareEnvForOperator(clientNamespace, Collections.singletonList(clientNamespace), Collections.emptyList());
+    }
+
+    protected void teardownEnvForOperator() {
```

Comment:
All these methods above require Javadoc: They're effectively API for the tests, so take the care to document the contract of that API.